Sources/MapboxNavigation/UserLocationStyle.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import MapboxMaps
+
+/**
+ A type that represents a `UIView` that is `CourseUpdatable`.
+ */
+public typealias UserCourseView = UIView & CourseUpdatable
+
+/**
+ The style of the user location indicator in a `NavigationMapView`.
+ */
+public enum UserLocationStyle {
+    /**
+     The course view representing the user’s location and course on the map, switching between `UserPuckCourseView` and `UserHaloCourseView` based on the level of location accuracy.
+     */
+    case courseView(configuration: UserCourseView)
+    
+    /**
+     A 2-dimensional puck from `MapboxMaps`. Optionally provide `Puck2DConfiguration` to configure the puck's appearance.
+     */
+    case puck2D(configuration: Puck2DConfiguration = Puck2DConfiguration())
```

Comment:
If there’s ever a need to distinguish between a customized 2D puck and an uncustomized one, accepting `nil` here would allow the SDK to make that distinction:

```swift
case puck2D(configuration: Puck2DConfiguration? = nil)
```

It would cause users of this case to need to handle `nil`, which may or may not be beneficial.

Reply:
Thanks for the suggestion, I'll improve this part here.